src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -354,5 +451,53 @@ private static VisualStudioInstance GetDevConsoleInstance()
             return null;
         }
 #endif
+
+        // Taken from MSBuild/NativeMethodsShared
+        private static readonly object IsRunningOnMonoLock = new object();
+
+        private static bool? _isRunningOnMono;
+
+        /// <summary>
+        /// Gets a flag indicating if we are running under MONO
+        /// </summary>
+        internal static bool IsRunningOnMono
+        {
+            get
+            {
+                if (_isRunningOnMono != null) return _isRunningOnMono.Value;
```

Comment:
nit:
```suggestion
                if (_isRunningOnMono.HasValue) return _isRunningOnMono.Value;
```

Reply:
this and the `IsRunningOnMono` are as-is from msbuild. I'll address the feedback though.